### DISCUSSION TITLE
Cannot build a project with `"noImplicitAny": true` when react-aria is a dependency

### DISCUSSION THREAD

**Discussion by VapidLinus** [adobe/react-spectrum#6417] [Q&A] [answered]
Heya, I'm looking for some help or feedback on whether I'm experiencing a bug or a configuration problem.

I have a NextJS project where I use react-aria as a dependency, and I run into a type error when I try to build:
```
./node_modules/@react-aria/button/src/useButton.ts:112:17
Type error: Parameter 'e' implicitly has an 'any' type.

  110 |       'aria-controls': props['aria-controls'],
  111 |       'aria-pressed': props['aria-pressed'],
> 112 |       onClick: (e) => {
      |                 ^
  113 |         if (deprecatedOnClick) {
  114 |           deprecatedOnClick(e);
  115 |           console.warn('onClick is deprecated, please use onPress');
```

I'm not sure why I get type errors in a dependency of mine. I have `node_modules` under `excluded` in tsconfig and I have `skipLibCheck: true`. But this appears to not be a bug but instead be [by design](https://stackoverflow.com/questions/42934675/why-tsc-shows-the-errors-on-dependency-module-typings#:~:text=The%20issue%20is%20marked%20as%20working%20as%20intended.).

I'm wondering if building with react-aria as a dependency is unsupported if I want to use `"noImplicitAny": true`? I'd really prefer to have it enabled.

This is my tsconfig:
```json
{
    "compilerOptions": {
        "target": "es2017",
        "lib": ["dom", "dom.iterable", "esnext"],
        "allowJs": true,
        "skipLibCheck": true,
        "strict": true,
        "forceConsistentCasingInFileNames": true,
        "noEmit": true,
        
        // TODO: We should strive towards this being true.
        //       react-aria has issues building with it being true for now.
        "noImplicitAny": true,

        "esModuleInterop": true,
        "module": "esnext",
        "moduleResolution": "node",
        "resolveJsonModule": true,
        "isolatedModules": true,
        "jsx": "preserve",
        "incremental": true,
        "noUncheckedIndexedAccess": true,
        "baseUrl": "src",
        "paths": {
            ...
        }
    },
    "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "**/*.cjs", "**/*.mjs"],
    "exclude": ["node_modules"]
}
```

**Comment by snowystinger** [comment, marked as answer]
Possibly will be fixed by working on https://github.com/adobe/react-spectrum/issues/3183
Otherwise can be discussed after that work is done. Things that will be asked are the same as https://github.com/adobe/react-spectrum/issues/1890#issuecomment-2252071558
